### PR TITLE
Accordion

### PR DESCRIPTION
**Pull request by aman-rana-vs**

### CODE REVIEW

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  withTiming,
+} from 'react-native-reanimated'
+import { DefaultStyle } from 'react-native-reanimated/lib/typescript/reanimated2/hook/commonTypes'
+import Icon from 'react-native-vector-icons/AntDesign'
+
+import { styles } from './accordion-styles'
+
+interface IAccordionProps {
+  /**accordionTitleStyle: is an optional prop which defines the style of the accordion title */
+  accordionTitleStyle?: DefaultStyle
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:16]: remove repeatitive "accordion" prefix, as its accordion component. Check all namings.

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  /**accordionStyle: is an optional prop which defines the style of the accordion */
+  accordionStyle?: DefaultStyle
+  /**children: is a required prop which dictates the children of the accordion */
+  children: JSX.Element
+  /**isDefaultOpen: is an optional prop which dictates the default state of accordion */
+  isDefaultOpen?: boolean
+  /**duration: is an optional prop which dictates the duration of animation */
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:23]: animation for what? make sure the comments have clear meaning.

File: AnimatedComponents/App.tsx [hunk 3]
```diff
@@ -2,91 +2,31 @@
...
-      },
-    }),
-    [],
-  )
-
-  const gradientWaveColor = useMemo(() => ['#1A63C5', '#1A63C5'], [])
-  const thumbColors = useMemo(() => ['#1A63C5', '#1A63C5'], [])
-
   return (
-    <SafeAreaView style={{ flex: 1 }}>
```
Comment (praveen-verma-vs) [inline AnimatedComponents/App.tsx:66]: Revert these changes, and confirm before making such changes, you can add your component at the last to showcase.

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  useDerivedValue,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated'
+import { DefaultStyle } from 'react-native-reanimated/lib/typescript/reanimated2/hook/commonTypes'
+import Icon from 'react-native-vector-icons/AntDesign'
+
+import { styles } from './accordion-styles'
+
+interface IAccordionProps {
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:14]: use prettier and try not to push code with typescript/ prettier issues.
https://eslint.org/docs/latest/rules/spaced-comment

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  isDefaultOpen?: boolean
+  /**duration: is an optional prop which dictates the duration of animation */
+  duration?: number
+  /**icon: is an optional prop which dictates the name of the icon */
+  icon?: string
+  /**title: is an optional prop which defines the title of the accordion */
+  title: string
+}
+
+const Accordion = (props: IAccordionProps) => {
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:31]: try creating components as "Accordion:React.FC<IAccordionProps>=(props)=>{....}"

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    accordionStyle = {},
+    children,
+    isDefaultOpen = false,
+    duration = 300,
+    icon = '',
+    title,
+  } = props
+
+  const open = useSharedValue(isDefaultOpen)
+  const height = useSharedValue(0)
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:43]: we try not to animate height/width instead use layout animations here- refer this
https://reactnative.dev/docs/layoutanimation

File: AnimatedComponents/App.tsx [hunk 7]
```diff
@@ -2,91 +2,31 @@
...
-      />
-      <IconTransitionButton
-        onPress={onIconTransition}
-        startIcon={<RightArrowIcon />}
-        failedIcon={<CrossIcon />}
-        successIcon={<TickIcon />}
-      />
-      <Slider type={SliderType.SingleValueSlider} sliderWidth={300} min={0} max={100} step={5} />
+    <SafeAreaView style={{ flex: 1, backgroundColor: '#91DDCF', gap: 10 }}>
+      <Accordion title="Lorem ipsum dolor">
```
Comment (praveen-verma-vs) [inline AnimatedComponents/App.tsx:12]: make the structure like this:

`<Accordion showSingleItemAtOnce  >
        <Accordion.Item id={0} title='Item 1'>
          <Content />
        </Accordion.Item>
 </Accordion> `
 
 => here we will use this optional prop: showSingleItemAtOnce- to open single item or multiple items at once.
 => <Accordion> - will be main wrapper

**Review by pvn013 — CHANGES_REQUESTED** [review]

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+  const rotateIconValue = useDerivedValue(() =>
+    withTiming(open ? '180deg' : '0deg', { duration: duration }),
+  )
+
+  const onPress = () => {
+    if (!isMultipleOpen) {
+      setActiveAccordionId(activeAccordionId === id ? undefined : id)
+    }
+    setOpen(!open)
+    LayoutAnimation.configureNext({
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:79]: check again if we require onpress animation and useEffect animation.

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+      duration: duration,
+      create: {
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:81]: we need not to create, we need to some other property of it please refer documentation.

File: AnimatedComponents/src/components/accordion/Accordion.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        type: LayoutAnimation.Types.easeIn,
+        property: LayoutAnimation.Properties.opacity,
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/accordion/Accordion.tsx:83]: here as well we don't need to alter "opacity" property, check animation again its not working at all.

**Review by pvn013 — CHANGES_REQUESTED** [review]
